pxemgr/pxemanager.go

```diff
@@ -187,77 +195,97 @@ func withSerialParam(serialHandler func(serial string, w http.ResponseWriter, r
+	mgr.apiRouter.Methods("PUT").PathPrefix("/admin/host/{serial}/set_provider_id").HandlerFunc(withSerialParam(mgr.setProviderId))
+	mgr.apiRouter.Methods("PUT").PathPrefix("/admin/host/{serial}/set_ipmi_addr").HandlerFunc(withSerialParam(mgr.setIPMIAddr))
+	mgr.apiRouter.Methods("PUT").PathPrefix("/admin/host/{serial}/set_cabinet").HandlerFunc(withSerialParam(mgr.setCabinet))
+	mgr.apiRouter.Methods("PUT").PathPrefix("/admin/host/{serial}/set_state").HandlerFunc(withSerialParam(mgr.setState))
+	mgr.apiRouter.Methods("PUT").PathPrefix("/admin/host/{serial}/set_etcd_cluster_token").HandlerFunc(withSerialParam(mgr.setEtcdClusterToken))
+	mgr.apiRouter.Methods("PUT").PathPrefix("/admin/host/{serial}/override").HandlerFunc(withSerialParam(mgr.override))
+
+	mgr.apiRouter.Methods("GET").PathPrefix("/admin/mayu_config").HandlerFunc(mgr.getMayuConfig)
+	mgr.apiRouter.Methods("PUT").PathPrefix("/admin/mayu_config").HandlerFunc(mgr.setMayuConfig)
+
+	mgr.apiRouter.Methods("GET").PathPrefix("/admin/hosts").HandlerFunc(mgr.hostsList)
 	// etcd discovery
 	if mgr.useInternalEtcdDiscovery {
-		etcdRouter := mgr.router.PathPrefix("/etcd").Subrouter()
+		etcdRouter := mgr.apiRouter.PathPrefix("/etcd").Subrouter()
 		mgr.defineEtcdDiscoveryRoutes(etcdRouter)
 	}
 
-	// boring stuff
-	mgr.router.Methods("GET").PathPrefix("/admin/hosts").HandlerFunc(mgr.hostsList)
-	mgr.router.Methods("GET").PathPrefix("/images/{serial}").HandlerFunc(mgr.imagesHandler)
-
-	// serve assets for yochu like etcd, fleet, docker, kubectl and rkt
-	mgr.router.PathPrefix("/yochu").Handler(http.StripPrefix("/yochu", http.FileServer(http.Dir(mgr.yochuPath))))
-
 	// add welcome handler for debugging
-	mgr.router.Path("/").HandlerFunc(mgr.welcomeHandler)
-
-	// serve static files like infopusher and mayuctl etc.
-	mgr.router.PathPrefix("/").Handler(http.FileServer(http.Dir(mgr.staticHTMLPath)))
+	mgr.apiRouter.Path("/").HandlerFunc(mgr.welcomeHandler)
 
 	glogWrapper := logging.NewGlogWrapper(8)
-	loggedRouter := handlers.LoggingHandler(glogWrapper, mgr.router)
+	loggedRouter := handlers.LoggingHandler(glogWrapper, mgr.apiRouter)
 
-	glog.V(8).Infoln(fmt.Sprintf("starting iPXE server at %s:%d", mgr.httpBindAddress, mgr.httpPort))
+	glog.V(8).Infoln(fmt.Sprintf("starting API server at %s:%d", mgr.bindAddress, mgr.apiPort))
 
 	if mgr.noTLS {
-		err := http.ListenAndServe(fmt.Sprintf("%s:%d", mgr.httpBindAddress, mgr.httpPort), loggedRouter)
+		err := http.ListenAndServe(fmt.Sprintf("%s:%d", mgr.bindAddress, mgr.apiPort), loggedRouter)
 		if err != nil {
-			return err
+			errChan <- mayuerror.MaskAny(err)
+			return
 		}
 	} else {
-		err := http.ListenAndServeTLS(fmt.Sprintf("%s:%d", mgr.httpBindAddress, mgr.httpPort), mgr.tlsCertFile, mgr.tlsKeyFile, loggedRouter)
+		err := http.ListenAndServeTLS(fmt.Sprintf("%s:%d", mgr.bindAddress, mgr.apiPort), mgr.tlsCertFile, mgr.tlsKeyFile, loggedRouter)
```

Comment:
Lets use https://golang.org/pkg/net/#JoinHostPort. This does the formatting magic out of the box.